### PR TITLE
WD-2205 Add Kubeflow beta 1.7 banner

### PR DESCRIPTION
**Pull request by carkod**
## Done

Sites tribe [got a request](https://docs.google.com/document/d/19_9cUiyjxWXPvLaZc0_DOglsCQwlJFW8gET7sZt7IYI/edit#) to add this Kubeflow 1.7 beta banner

This banner is scheduled to be up on 8th of March and be taken down on 22nd of March (check the brief linked previously).

I've also fixed your dotrun issue, I think it just needs some time to be able to `curl`, dotrun is probably doing stuff

## How to QA

- Navigate to https://charmhub-io-1529.demos.haus/kubeflow
- Click on the links.

## Issue / Card
Fixes https://warthogs.atlassian.net/browse/WD-2205?atlOrigin=eyJpIjoiNGMzNjRhODYxZDY5NDFhMzhlMDllMzExMzE0MzRjODgiLCJwIjoiaiJ9

## Screenshots

image

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -152,3 +153,19 @@ def site_map_operators():
...
+        start = datetime.strptime(start_date, "%Y-%m-%d")
+        present = datetime.now()
+        print("hello", start, present)
+        return start <= present < end
+    except ValueError:
+        return False
+
+
+@app.context_processor
+def utility_processor():
```
Comment (Lukewh) [inline webapp/app.py:170]: Can you move these into `handlers.py`, where `utility_processor` is already defined? I'd like to keep `app.py` as clean as possible, thanks.

**Review by Lukewh — CHANGES_REQUESTED** [review]

**Comment by Lukewh** [comment]
If I click on the "configure" tab, the banner disappears, should it be on all kubeflow pages?

**Comment by Lukewh** [comment]
Please don't merge this, I want to talk to Andreea to make sure this is the right solution to what they're trying to achieve

**Comment by carkod** [comment]
> Please don't merge this, I want to talk to Andreea to make sure this is the right solution to what they're trying to achieve

Good point. I can also make it work with whatever starts with "kubeflow". Let me know

**Review by Lukewh — APPROVED** [review]
:+1: LGTM

**Comment by Lukewh** [comment]
@carkod thanks for updating, and this looks good now - you can change the start date and get it merged, thanks :)

**Comment by carkod** [comment]
> @carkod thanks for updating, and this looks good now - you can change the start date and get it merged, thanks :)

Thanks. I assume you got confirmation that they want it displayed in all /kubeflow/<path> pages?

**Comment by Lukewh** [comment]
I see you're still working on this, let me know when to review again

**Comment by carkod** [comment]
> I see you're still working on this, let me know when to review again

Sorry, I'm just trying to fix the dotrun issue. It started failing again, but I can see what the problem is. It's doing the `curl` and then failing, but after a while the assets are built...

Also it looks a bit weird to me, that you have dotrun as required but run-image is not required? I assume you don't `docker run` your PRs, so it'd be good to have this check as required, because if that's broken, very likely prod or staging would be broken.

**Comment by Lukewh** [comment]
> > I see you're still working on this, let me know when to review again
> 
> Sorry, I'm just trying to fix the dotrun issue. It started failing again, but I can see what the problem is. It's doing the `curl` and then failing, but after a while the assets are built...
> 
> Also it looks a bit weird to me, that you have dotrun as required but run-image is not required? I assume you don't `docker run` your PRs, so it'd be good to have this check as required, because if that's broken, very likely prod or staging would be broken.
> 
> But anyway, it's not work of this specific feature, I could leave it for another PR

Yeah, if things work without the changes to dotrun, move them to another PR

**Comment by carkod** [comment]
> > > I see you're still working on this, let me know when to review again
> > 
> > 
> > Sorry, I'm just trying to fix the dotrun issue. It started failing again, but I can see what the problem is. It's doing the `curl` and then failing, but after a while the assets are built...
> > Also it looks a bit weird to me, that you have dotrun as required but run-image is not required? I assume you don't `docker run` your PRs, so it'd be good to have this check as required, because if that's broken, very likely prod or staging would be broken.
> > But anyway, it's not work of this specific feature, I could leave it for another PR
> 
> Yeah, if things work without the changes to dotrun, move them to another PR

Ok, seems like it worked!

Everything else is the same, I just changed the `pr.yaml` command

**Review by Lukewh — APPROVED** [review]
:+1: LGTM

**Pull request merged**